ops/model.py

```diff
@@ -107,12 +107,14 @@ class Model:
     from any class that derives from Object.
     """
 
-    def __init__(self, meta: 'ops.charm.CharmMeta', backend: '_ModelBackend'):
+    def __init__(self, meta: 'ops.charm.CharmMeta', backend: '_ModelBackend',
+                 breaking_relation_id: Optional[int] = None):
```

Comment:
I think it would be more elegant if we attached this information to the event instead of the model.

breaking-relation-id is deceptive as it suggests any relation ID could be that of a relation that is breaking, while in fact the only relation we could possibly know to be breaking is 'the current one', aka 'the one the current event is about'.
So it makes more sense to me to attach that information to the event.

That also spares us to add model.relation.get_active because there's only ever going to be one relation in breaking state: the one we get from the event we're observing.

I propose the following api:

```
def _on_foo_relation_event(self, e:RelationEvent):
    if e.is_dying:
        ...
```

this feels a bit more `ops`y.

Reply:
that said, this would make it more complicated to access the same information from code that doesn't directly have access to the event object. Maybe this was the whole point of this exercise?

Reply:
The motivation for me with this issue is a case where, if possible, not being tied to the event would be nice.  I had tried to implement a [nicely encapsulated getter](https://github.com/canonical/kubeflow-dashboard-operator/pull/118/files#diff-5af16278717441c62aff8728f4b371e49c0e8a8e3560fac251562b7a8a6d1258R161) on a relation charm lib, but found since it might be called during `relation-broken` I needed to spot and omit the breaking application.  To do that existing `ops` and without spying on the `JUJU_*` variables, it meant passing the `event` through several layers of code down to the getter.  Not impossible, but it felt awkward at best

Reply:
You're right though that this is event-specific behaviour.  But then since `model.relations` exposes the relation information and the event influences what that data means, I think `ops` sort of commits itself to giving access to the full context via `model.relations`

Reply:
Yeah, I agree with Andrew here: we specifically want this info to be available on the `Relation` object, rather then the event.